src/rust/graph-query-service/src/property_query.rs

```diff
@@ -0,0 +1,154 @@
+    pub value: String,
+}
+
+// We should push our filtering logic into here
+
+#[derive(Clone)]
+pub struct PropertyQueryExecutor {
+    scylla_client: Arc<CachingSession>,
+}
+
+impl PropertyQueryExecutor {
+    pub fn new(scylla_client: Arc<CachingSession>) -> Self {
+        Self { scylla_client }
+    }
+
+    pub async fn get_immutable_string(
+        &self,
+        tenant_id: uuid::Uuid,
+        uid: Uid,
+        property_name: &PropertyName,
+    ) -> Result<Option<StringField>, PropertyQueryError> {
+        let tenant_urn = tenant_id.simple();
+
+        let mut query = scylla::query::Query::from(format!(
+            r"
+            SELECT value
+            FROM tenant_keyspace_{tenant_urn}.immutable_strings
+            WHERE
+                uid = ? AND
+                populated_field = ?
+            LIMIT 1
+            ALLOW FILTERING;
+            "
+        ));
+
+        query.set_is_idempotent(true);
+
+        let query_result = self
+            .scylla_client
+            .execute(query, &(uid.as_i64(), &property_name.value))
+            .await?;
+
+        let row = match query_result.maybe_first_row_typed::<(String,)>()? {
+            Some((row,)) => row,
+            None => return Ok(None),
+        };
+
+        Ok(Some(StringField {
+            uid,
+            populated_field: property_name.clone(),
+            value: row,
+        }))
+    }
+
+    pub async fn get_edges(
+        &self,
+        tenant_id: uuid::Uuid,
+        uid: Uid,
+        edge_name: &EdgeName,
+    ) -> Result<Option<Vec<EdgeRow>>, PropertyQueryError> {
+        let tenant_urn = tenant_id.simple();
+
+        let mut query = scylla::query::Query::from(format!(
+            r"
+            SELECT r_edge_name, destination_uid
+            FROM tenant_keyspace_{tenant_urn}.edge
+            WHERE
+                source_uid = ? AND
+                f_edge_name = ?
+            ALLOW FILTERING;
+            "
+        ));
+
+        println!("query: \n{}\n", &query.contents);
+
+        query.set_is_idempotent(true);
+
+        let query_result = self
+            .scylla_client
+            .execute(query, &(uid.as_i64(), &edge_name.value))
+            .await?;
+
+        let rows = query_result.rows_typed_or_empty::<(String, i64)>();
+
+        let mut edge_rows = Vec::new();
+        for row in rows {
+            let (r_edge_name, destination_uid) = row?;
+            edge_rows.push(EdgeRow {
+                source_uid: uid,
+                f_edge_name: edge_name.clone(),
+                r_edge_name: EdgeName::try_from(r_edge_name)
+                    .map_err(PropertyQueryError::InvalidStoredEdgeName)?,
+                destination_uid: Uid::from_i64(destination_uid)
+                    .ok_or(PropertyQueryError::InvalidUidInDb(
+                        destination_uid,
+                        format!("source_uid: {uid:?} f_edge_name: {edge_name:?} invalid destination_uid {destination_uid}"),
```

Comment:
This as the "reason" for `PropertyQueryError::InvalidUidInDb` looks really weird to me, the variant has a field `destination_uid`, but it's then mixing other fields in this string that is later formatted again (`#[error("InvalidUid: {0} reason: {1}")]`).

It looks to me like PropertyQueryError::InvalidUidInDb should be more along the lines of:
```
#[error("invalid destination_uid '{destination_uid}': source_uid: '{source_uid:?}': f_edge_name: '{f_edge_name}'")]
PropertyQueryError::InvalidUidInDb {
    destination_uid: Uid,
    source_uid: Uid,
    f_edge_name: String,
}
```
where this line is then something like
```
.ok_or(PropertyQueryError::InvalidUidInDb { destination_uid, uid, edge_name: edge_name.to_string() })
```